File: scripts/extractBareme.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import { createRequire } from 'module';

const require = createRequire(import.meta.url);
const pdfParseModule = require('pdf-parse');
const pdfParse = pdfParseModule.default || pdfParseModule;

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function extractBaremeData() {
    try {
        const pdfPath = path.join(__dirname, '..', 'BAREME AT-MP.pdf');
        const dataBuffer = fs.readFileSync(pdfPath);
        
        console.log('📄 Extraction du contenu du barème PDF...\n');
        
        const data = await pdfParse(dataBuffer);
        
        // Sauvegarde du texte complet
        const outputPath = path.join(__dirname, '..', 'bareme_extracted.txt');
        fs.writeFileSync(outputPath, data.text, 'utf8');
        
        console.log(`✅ Extraction terminée !`);
        console.log(`📊 Pages: ${data.numpages}`);
        console.log(`📝 Caractères extraits: ${data.text.length}`);
        console.log(`💾 Fichier sauvegardé: bareme_extracted.txt\n`);
        
        // Afficher un aperçu
        console.log('--- APERÇU DU CONTENU ---');
        console.log(data.text.substring(0, 2000));
        console.log('\n... (voir bareme_extracted.txt pour le contenu complet)');
        
    } catch (error) {
        console.error('❌ Erreur lors de l\'extraction:', error.message);
    }
}

extractBaremeData();

File: scripts/extract_bareme.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Script d'extraction du barème AT-MP depuis le PDF
"""

import sys
import os

try:
    import PyPDF2
    print("✅ PyPDF2 installé")
except ImportError:
    print("❌ PyPDF2 non installé. Installation en cours...")
    os.system(f"{sys.executable} -m pip install PyPDF2")
    import PyPDF2

def extract_pdf_text(pdf_path):
    """Extrait le texte du PDF"""
    print(f"📄 Ouverture du PDF: {pdf_path}")
    
    with open(pdf_path, 'rb') as file:
        reader = PyPDF2.PdfReader(file)
        num_pages = len(reader.pages)
        
        print(f"📊 Nombre de pages: {num_pages}")
        
        full_text = ""
        for i, page in enumerate(reader.pages, 1):
            print(f"⏳ Extraction page {i}/{num_pages}...", end='\r')
            text = page.extract_text()
            full_text += f"\n\n=== PAGE {i} ===\n\n{text}"
        
        print(f"\n✅ Extraction terminée ! Total: {len(full_text)} caractères")
        
        return full_text

def main():
    # Chemin du PDF
    pdf_path = os.path.join(os.path.dirname(__file__), '..', 'BAREME AT-MP.pdf')
    
    if not os.path.exists(pdf_path):
        print(f"❌ Fichier PDF non trouvé: {pdf_path}")
        return
    
    # Extraction
    text = extract_pdf_text(pdf_path)
    
    # Sauvegarde
    output_path = os.path.join(os.path.dirname(__file__), '..', 'bareme_extracted.txt')
    with open(output_path, 'w', encoding='utf-8') as f:
        f.write(text)
    
    print(f"💾 Fichier sauvegardé: {output_path}")
    
    # Afficher un aperçu
    print("\n" + "="*60)
    print("APERÇU DU CONTENU (1000 premiers caractères):")
    print("="*60)
    print(text[:1000])
    print("\n... (voir bareme_extracted.txt pour le contenu complet)")

if __name__ == "__main__":
    main()
